webapp/sources/rudder/rudder-rest/src/main/scala/com/normation/rudder/Authorizations.scala

```diff
@@ -330,62 +330,78 @@ sealed trait Role {
-  }
-  final case object Workflow   extends Role {
-    val name = "workflow"; def rights = Rights(ua ++ A.workflowKind ++ A.complianceKind)
-  }
-  final case object Deployer   extends Role {
-    val name = "deployer"; def rights = Rights(ua ++ A.Deployer.values ++ A.complianceKind)
-  }
-  final case object Validator  extends Role {
-    val name = "validator"; def rights = Rights(ua ++ A.Validator.values ++ A.complianceKind)
-  }
-  case object Configuration    extends Role {
-    val name = "configuration"; val rights = Rights(ua ++ A.configurationKind.map(identity))
+    override def debugString = s"authz[${rights.displayAuthorizations}]"
+
+    override def addAuthz(authz: Set[A]): Role = Custom(Rights(rights.authorizationTypes ++ authz))
   }
-  final case object ReadOnly   extends Role { val name = "read_only"; def rights = Rights(ua ++ allRead)                      }
-  final case object Compliance extends Role { val name = "compliance"; def rights = Rights(ua ++ A.complianceKind)            }
-  final case object Inventory  extends Role { val name = "inventory"; def rights = Rights(ua ++ Set(A.Node.Read))             }
-  final case object RuleOnly   extends Role {
-    val name = "rule_only"; def rights = Rights(ua ++ Set(A.Configuration.Read, A.Rule.Read))
+
+  // built-in roles
+  final case class Builtin(name: String, rights: Rights) extends Role {
+    override def addAuthz(authz: Set[A]): Role = {
+      Builtin(name, Rights(rights.authorizationTypes ++ authz))
+    }
   }
 
-  // a special Role that means that a user has no rights at all. That role must super-seed any other right given by other roles
-  final case object NoRights extends Role { val name = "no_rights"; def rights = Rights(Set(AuthorizationType.NoRights)) }
+  // a special account, with all rights, present and future, even if declared at runtime.
+  final case object Administrator extends Role {
+    val name = "administrator";
 
-  // this is the anonymous custom roles, the one computed on fly for user who have several roles in their attribute
-  final case class Custom(rights: Rights) extends Role {
-    val name                 = "custom"
-    override def debugString = s"authz[${rights.displayAuthorizations}]"
+    def rights = Rights.AnyRights
+
+    // no-op
+    override def addAuthz(authz: Set[A]): Role = this
   }
-  def forRight(right: AuthorizationType) = Custom(Rights.forAuthzs(right))
+
+  // a special Role that means that a user has no rights at all. That role must super-seed any other right given by other roles
+  final case object NoRights extends Role {
+    val name = "no_rights";
+
+    def rights = Rights(Set(A.NoRights))
+
+    // no-op
+    override def addAuthz(authz: Set[A]): Role = this
+  }
+
+  // standard predefined built-in roles
+  def standardBuiltIn: Map[String, Role] = List(
+    Builtin("user", Rights(ua ++ A.nodeKind ++ A.configurationKind)),
```

Comment:
I think all built in should have been objects that extends BuiltIn not instances of BuiltIn so that we can find their key more easily in code 

ie:

object ReadOnly extends BuiltIn("read_only", Rights(ua ++ A.allKind.collect { case x: ActionType.Read => x }))


and in plugin:
instead of redifining a builtIn

 Builtin("read_only", Rights(Set(right1, right2)))

you can do

ReadOnly.addRights(Set(right1,right2))

Reply:
addAuthz)

Reply:
I thought to do it, but I have a problem with that: if you add several time rights, with the object case you (at least me) would be ver very tempted to do `ReadOnly.addRights(Set(right1,right2))`. But the `read_only` was already modified, so you must have done a look-up on `read_only` (but since the object is there, it is not what is on least resistance path). 

So the alternative I see are:
- base built-in roles are stateful object, their rights are mutable in a `Ref` (to avoid parrallel pb). I almost did that but then base built in roles are more special than plugins, it breaks simitry, and there was a lot of boilerplate added (`getRights` is `ZIO` etc). But perhaps it just need more polishing
- what is stable is just the name/a typed version of the name (ie `val READ_ONLY = "read_only"` or `object ReadOnly extends BuiltinRoleName("read_only")` for ex) and we keep the remaning parts identical, just lookup identifier are from stable values in place of a string that can have typo.

Reply:
In fact as stated here: https://github.com/Normation/rudder/pull/5004/files#r1306393613 i think registerBuiltIn should extend existing Role, not replace them. so you don't have to fetch them beforehand.

Maybe it's just a role identifier and list of authz granted (added) to that role to pass as parameter to registerBuiltIn. 

I think i really want at least an object with builtIn identifiers : 

object BuildInRole {

val READ_ONLY = "read_only" 

}

(whatever the case, i would prefere lowercase, but i think you would prefer uppercase

Reply:
In fact as stated here: https://github.com/Normation/rudder/pull/5004/files#r1306393613 i think registerBuiltIn should extend existing Role, not replace them. so you don't have to fetch them beforehand.

Maybe it's just a role identifier and list of authz granted (added) to that role to pass as parameter to registerBuiltIn. 

I think i really want at least an object with builtIn identifiers : 

object BuildInRole {

val READ_ONLY = "read_only" 

}

(whatever the case, i would prefere lowercase, but i think you would prefer uppercase

Reply:
OK, I think I understand what you meant, coding an update.